fcli-core/fcli-fod/src/main/java/com/fortify/cli/fod/scan/cli/cmd/FoDScanStartDastCommand.java

```diff
@@ -13,47 +13,48 @@
-import java.time.format.DateTimeFormatter;
-import java.util.Properties;
-
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fortify.cli.common.output.transform.IActionCommandResultSupplier;
 import com.fortify.cli.common.output.transform.IRecordTransformer;
 import com.fortify.cli.common.progress.cli.mixin.ProgressWriterFactoryMixin;
 import com.fortify.cli.common.util.FcliBuildPropertiesHelper;
-import com.fortify.cli.common.util.StringUtils;
 import com.fortify.cli.fod._common.cli.mixin.FoDDelimiterMixin;
 import com.fortify.cli.fod._common.output.cli.AbstractFoDJsonNodeOutputCommand;
 import com.fortify.cli.fod._common.output.mixin.FoDOutputHelperMixins;
 import com.fortify.cli.fod._common.util.FoDEnums;
+import com.fortify.cli.fod.assessment_type.helper.FoDAssessmentTypeDescriptor;
+import com.fortify.cli.fod.assessment_type.helper.FoDAssessmentTypeHelper;
+import com.fortify.cli.fod.entitlement.helper.FoDEntitlementHelper;
+import com.fortify.cli.fod.entitlement.helper.FoDInvalidEntitlementException;
 import com.fortify.cli.fod.release.cli.mixin.FoDReleaseByQualifiedNameOrIdResolverMixin;
-import com.fortify.cli.fod.release.helper.FoDReleaseDescriptor;
-import com.fortify.cli.fod.scan.cli.mixin.FoDEntitlementPreferenceTypeMixins;
+import com.fortify.cli.fod.scan.cli.mixin.FoDEntitlementFrequencyTypeMixins;
 import com.fortify.cli.fod.scan.cli.mixin.FoDInProgressScanActionTypeMixins;
 import com.fortify.cli.fod.scan.cli.mixin.FoDRemediationScanPreferenceTypeMixins;
-import com.fortify.cli.fod.scan.helper.FoDAssessmentType;
-import com.fortify.cli.fod.scan.helper.FoDScanAssessmentTypeDescriptor;
-import com.fortify.cli.fod.scan.helper.FoDScanDescriptor;
 import com.fortify.cli.fod.scan.helper.FoDScanHelper;
 import com.fortify.cli.fod.scan.helper.FoDScanType;
 import com.fortify.cli.fod.scan.helper.dast.FoDScanDastHelper;
 import com.fortify.cli.fod.scan.helper.dast.FoDScanDastStartRequest;
 import com.fortify.cli.fod.scan_config.helper.FoDScanConfigDastDescriptor;
 import com.fortify.cli.fod.scan_config.helper.FoDScanConfigDastHelper;
-
 import kong.unirest.UnirestInstance;
 import lombok.Getter;
 import picocli.CommandLine.Command;
 import picocli.CommandLine.Mixin;
 import picocli.CommandLine.Option;
 
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Properties;
+
 @Command(name = FoDOutputHelperMixins.StartDast.CMD_NAME)
 public class FoDScanStartDastCommand extends AbstractFoDJsonNodeOutputCommand implements IRecordTransformer, IActionCommandResultSupplier {
     DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm");
     @Getter @Mixin private FoDOutputHelperMixins.StartDast outputHelper;
     @Mixin private FoDDelimiterMixin delimiterMixin; // Is automatically injected in resolver mixins
     @Mixin private FoDReleaseByQualifiedNameOrIdResolverMixin.PositionalParameter releaseResolver;
+    @Option(names = {"--assessment-type"}, required = true)
```

Comment:
Ideally, for consistency with the start-sast command, the start-dast command should assume the scan has already been configured (I think that includes settings like assessment type and entitlement id), and simply copy the existing settings from the scan-setup, rather than providing this as options on the start-dast command. From a user perspective, it should look like this command invokes the 'scan-start-with-defaults` endpoint, even though that doesn't exist (yet) for DAST.

Reply:
Yes it should do, however I'm not sure we can get all the information from the API and might take some time to implement.

I would suggest "hiding" the command for now (2.0.x release?)

Reply:
Yes, hiding this for now would probably be easiest & best